src/distilabel/steps/tasks/magpie/generator.py

```diff
@@ -209,35 +208,6 @@ class MagpieGenerator(GeneratorTask, MagpieBase):
         default=None, description="The number of rows to generate."
     )
 
-    def model_post_init(self, __context: Any) -> None:
-        """Checks that the provided `LLM` uses the `MagpieChatTemplateMixin`."""
-        super().model_post_init(__context)
-
-        if not isinstance(self.llm, MagpieChatTemplateMixin):
-            raise ValueError(
-                f"`Magpie` task can only be used with an `LLM` that uses the `MagpieChatTemplateMixin`."
-                f"`{self.llm.__class__.__name__}` doesn't use the aforementioned mixin."
-            )
-
-        self.llm.use_magpie_template = True
```

Comment:
@gabrielmbmb was there a reason you were defining these again? They were also defined in the parent class.

Reply:
Yes, having the `model_post_init` in the `MagpieBase` class and then inheriting in `Magpie` and `MagpieGenerator` didn't trigger it, so that's why it's duplicate in `Magpie` and `MagpieGenerator` tasks. Apparently, this is an issue with pydantic that should be fixed but it's not working and didn't want to expend too much time so I decided to duplicate the code.